ros_msft_onnx/CMakeLists.txt

```diff
@@ -25,47 +35,74 @@ find_package(catkin REQUIRED COMPONENTS
   tf
 )
 
-get_cmake_property(_variableNames VARIABLES)
-foreach (_variableName ${_variableNames})
-  #message(STATUS "${_variableName}=${${_variableName}}")
-endforeach()
-
-
 catkin_package(
   INCLUDE_DIRS include
-  CATKIN_DEPENDS std_msgs geometry_msgs visualization_msgs ros_msft_onnx_msgs image_transport roscpp cv_bridge tf onnxruntime_vendor
+  CATKIN_DEPENDS std_msgs geometry_msgs visualization_msgs ros_msft_onnx_msgs image_transport roscpp cv_bridge tf
+  CFG_EXTRAS  "onnxruntime_vendor-extras.cmake"
 )
 
 include_directories(
   include
   ${catkin_INCLUDE_DIRS}
   ${EIGEN3_INCLUDE_DIRS}
-  ${onnxruntime_vendor_INCLUDE_DIRS}
 )
 
 add_executable(${PROJECT_NAME}_node src/ros_msft_onnx.cpp src/main.cpp src/yolo_box.cpp src/pose_parser.cpp)
-add_dependencies(${PROJECT_NAME}_node ${${PROJECT_NAME}_EXPORTED_TARGETS} ${catkin_EXPORTED_TARGETS} ${onnxruntime_vendor_TARGETS})
-target_link_libraries(${PROJECT_NAME}_node ${catkin_LIBRARIES} ${OpenCV_LIBRARIES} ${onnxruntime_vendor_TARGETS})
+add_dependencies(${PROJECT_NAME}_node ${catkin_EXPORTED_TARGETS})
+target_link_libraries(${PROJECT_NAME}_node ${catkin_LIBRARIES} ${OpenCV_LIBRARIES} ${EIGEN3_LIBS})
+
+message("Installing onnxruntime_vendor Nuget package")
+
+if(CUDA_SUPPORT)
+  set(PACKAGES_CONFIG "packages.gpu.config")
+  set(ONNX_RUNTIME "Microsoft.ML.OnnxRuntime.Gpu.1.4.0")
+else()
+  set(PACKAGES_CONFIG "packages.cpu.config")
+  set(ONNX_RUNTIME "Microsoft.ML.OnnxRuntime.1.4.0")
+endif()
+
+if(MSVC)
+  set(ARCH "win-x64" )
+  find_program(NUGET nuget PATHS C:\\opt\\ros\\noetic\\x64\\bin)
+
+  if(NOT NUGET)
+    message(FATAL "CMake could not find the nuget command line tool. Please install it from nuget.org")
+  else()
+    execute_process(COMMAND ${NUGET} restore ${CMAKE_CURRENT_SOURCE_DIR}/src/${PACKAGES_CONFIG} -PackagesDirectory ${CMAKE_CURRENT_BINARY_DIR})
```

Comment:
can you change this to use tar on both windows and linux in the ROS2 version, so that we don't have to install extra tooling?

Reply:
made the suggested changes. Can you review ?